src/audio/module_adapter/module_adapter.c

```diff
@@ -633,31 +657,37 @@ int module_adapter_copy(struct comp_dev *dev)
 
 		/* copy source samples into input buffer */
 		list_for_item(blist, &dev->bsource_list) {
+			struct comp_buffer __sparse_cache *src_c;
 			uint32_t bytes_to_process;
 			int frames, source_frame_bytes;
 
 			source = container_of(blist, struct comp_buffer, sink_list);
-			source_c = buffer_acquire(source);
+			src_c = buffer_acquire(source);
+
+			/* check if the source dev is in the same state as the dev */
+			if (src_c->source && src_c->source->state != dev->state)
+				continue;
```

Comment:
did I comment on a different location yesterday? Here too - proceeding with processing `if (src_c->source == NULL)` doesn't seem right. Probably this should be `if (!src_c->source || src_c->source->state != dev->state)`